examples/hello_world/README.md

```diff
@@ -1,194 +0,0 @@
-(e.g., `impl HelloWorldDatabase`). The second is the key -- in this
-simple example, that's just `()`.
-
-**Invoking a query.** In the first line of the function we see how to
-invoke a query for a given key:
-
-```rust
-let input_string = db.input_string(());
-```
-
-You simply call the function and give the key you want -- in this case
-`()`.
-
-### Step 3: Define the database struct
-
-The final step is to create the **database struct** which will
-implement the traits from each of your query groups. This struct
-combines all the parts of your system into one whole; it can also add
-custom state of your own (such as an interner or configuration). In
-our simple example though we won't do any of that. The only field that
-you **actually** need is a reference to the **salsa runtime**; then
-you must also implement the `salsa::Database` trait to tell salsa
-where to find this runtime:
-
-```rust
-#[derive(Default)]
-struct DatabaseStruct {
-    runtime: salsa::Runtime<DatabaseStruct>,
-}
-
-impl salsa::Database for DatabaseStruct {
-    fn salsa_runtime(&self) -> &salsa::Runtime<DatabaseStruct> {
-        &self.runtime
-    }
-}
-```
-
-Next, you must use the `database_storage!` to specify the set of query
-groups that your database stores. This macro generates the internal
-storage struct used to store your data. To use the macro, you
-basically list out all the traits:
-
-```rust
-salsa::database_storage! {
-    DatabaseStruct { // <-- name of your context type
-        impl HelloWorldDatabase;
-    }
-}
-```
-
-The `database_storage` macro will also implement the
-`HelloWorldDatabase` trait for your query context type.
-
-**Use the database.** Now that we've defined our database, we can
-start using it:
-
-```rust
-fn main() {
-    let mut db = DatabaseStruct::default();
-
-    println!("Initially, the length is {}.", db.length().get(()));
-
-    db.query_mut(InputString)
-        .set((), Arc::new(format!("Hello, world")));
-
-    println!("Now, the length is {}.", db.length().get(()));
-}
-```
-
-One thing to notice here is how we set the value for an input query:
-
-```rust
-    db.query_mut(InputString)
-        .set((), Arc::new(format!("Hello, world")));
-```
-
-The `db.query_mut(Foo)` method takes as argument the query type that
-characterizes your query. It gives back a "mutable query table" type,
-which lets you invoke `set` to set the value of an input query. There
-is also a `query` method that gives access to other advanced methods
-in fact, the standard call `db.query_name(key)` to access a query is
-just a shorthand for `db.query(QueryType).get(key)`.
-
-Finally, if we run this code:
-
-```bash
-> cargo run --example hello_world
-   Compiling salsa v0.2.0 (/Users/nmatsakis/versioned/salsa)
-    Finished dev [unoptimized + debuginfo] target(s) in 0.94s
-     Running `target/debug/examples/hello_world`
-Initially, the length is 0.
-Now, the length is 12.
-```
-
-Amazing.
-
```

Comment:
ah, the constant struggle between "let's write awesome guide!" and "oh noes, everything's changed, the guide is completely obsolete" :) 

Seems ok to keep most of the guide in the doc comments, until we've mostly frozen the design and started with that book.

Reply:
Heh, yes. I felt like the README wasn't adding enough value to make it worthwhile, and that ultimately we want a book.